### ISSUE TITLE
Popover inside Sheet or Modal is not accessible with Voiceover + Safari

### ISSUE THREAD

**Issue by braheem** [Shopify/polaris#4285]
## Issue summary

When using a popover inside of a polaris Sheet or Modal component, voiceover in safari gets stuck on the activator button. Voiceover focus (different from dom focus) stays on the activator button, which prevents the popover content from being interacted with. This is a critical bug occurring voiceover + safari which is the most common combination for voiceover users.

## Expected behavior

The voiceover focus should move to the popover content.

## Actual behavior

The voiceover focus stays on the activator button for the popover.

## Steps to reproduce the problem

1. Nest a popover inside of a polaris sheet or modal component
1. In voiceover on safari, try to interact with the popover content

## Reduced test case

This is reproducible in this sandbox https://codesandbox.io/s/vigilant-easley-ie7m1?file=/App.js

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: latest
- Browser: Safari
- Device: Macbook pro
- Operating System: Big Sur 11.4
